gcloud/logging/logging_service_v2_api.py

```diff
@@ -0,0 +1,245 @@
+# and updates to that file get reflected here through a refresh process.
+# For the short term, the refresh process will only be runnable by Google engineers.
+# Manual additions are allowed because the refresh process performs
+# a 3-way merge in order to preserve those manual additions. In order to not
+# break the refresh process, only certain types of modifications are
+# allowed.
+#
+# Allowed modifications:
+# 1. New methods (these should be added to the end of the class)
+#
+# Happy editing!
+
+from google.api import monitored_resource_pb2
+from google.gax import api_callable
+from google.gax import api_utils
+from google.gax import page_descriptor
+from google.logging.v2 import log_entry_pb2
+from google.logging.v2 import logging_pb2
+
+class LoggingServiceV2Api(object):
+    """Service for ingesting and querying logs."""
+
+    # The default address of the logging service.
+    _SERVICE_ADDRESS = 'logging.googleapis.com'
+
+    # The default port of the logging service.
+    _DEFAULT_SERVICE_PORT = 443
+
+    # The scopes needed to make gRPC calls to all of the methods defined in
+    # this service
+    _ALL_SCOPES = (
+        'https://www.googleapis.com/auth/cloud-platform',
+        'https://www.googleapis.com/auth/cloud-platform.read-only',
+        'https://www.googleapis.com/auth/logging.admin',
+        'https://www.googleapis.com/auth/logging.read',
+        'https://www.googleapis.com/auth/logging.write'
+    )
+
+    def __init__(
+            self,
+            service_path=_SERVICE_ADDRESS,
+            port=_DEFAULT_SERVICE_PORT,
+            channel=None,
+            ssl_creds=None,
+            scopes=_ALL_SCOPES,
+            is_idempotent_retrying=True,
+            max_attempts=3,
+            timeout=30):
+        self.defaults = api_callable.ApiCallableDefaults(
+            timeout=timeout,
+            max_attempts=max_attempts,
+            is_idempotent_retrying=is_idempotent_retrying)
+
+        self.stub = api_utils.create_stub(
+            logging_pb2.beta_create_LoggingServiceV2_stub,
+            service_path,
+            port,
+            ssl_creds=ssl_creds,
+            channel=channel,
+            scopes=scopes)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+
+    def close(self):
+        del self.stub
+
+    # Page descriptors
+    _LIST_LOG_ENTRIES_DESCRIPTOR = page_descriptor.PageDescriptor(
+        'page_token',
+        'next_page_token',
+        'entries')
+    _LIST_MONITORED_RESOURCE_DESCRIPTORS_DESCRIPTOR = page_descriptor.PageDescriptor(
+        'page_token',
+        'next_page_token',
+        'resource_descriptors')
+
+    # Service calls
+    def delete_log(
+            self,
+            log_name='',
+            **kwargs):
+        """
+        Deletes a log and all its log entries.
+        The log will reappear if it receives new entries.
```

Comment:
A docstring should have a space between the first line and the rest. The first should be a short description. See [PEP257][1].

[1]: https://www.python.org/dev/peps/pep-0257/